### ISSUE TITLE
Replace 'findWhere' for 'where' and return BaseRepository instead of a Collection

### ISSUE THREAD

**Issue by pjmartorell** [andersao/l5-repository#192]
My suggestion is to rename methods `findWhere`, `findWhereIn` and `findWhereNotIn` to `where`, `whereIn` and `whereNotIn` and return the `BaseRepository` instead of a `Collection` of results. That would be more consistent with how Eloquent ORM defines these methods and would be more intuitive for users.

Applying these changes would allow, for example, paginate results, calling `$this->postRepository->findWhere([user_id => 1])->paginate(25)`. Now it's not possible to do that without chaining a `scopeQuery` (#129), which in my opinion is really clumsy and breaks with the repository pattern, since you are inserting code to directly query the ORM.

What do you think?

**Comment by pjmartorell** [comment]
Well, the problem here is that the repository pattern states that repositories' methods should return Collections but in this package there's a mix of methods that return a `Collection` object and methods that return a `BaseRepository` object to allow chaining more methods.

**Comment by enrichit** [comment]
Perhaps bundle a pagination criteria with the library as suggested in #129? I'm going to give it a try.

**Comment by rbruhn** [comment]
I've been running into this problem a lot. Not only with paginate but other methods as well. I simply started created Criteria for what I needed instead of fighting with it. I agree with @pjmartorell there should be default Criteria included with the package to move these methods out of the BaseRepository. For example:

https://gist.github.com/rbruhn/944d507c07590c8cbe8dfcbc1ca25cfd
I needed the above to select a record and returning the first one found:
```
$this->workflowManager->pushCriteria(new WhereFields(['expedition_id' => $expeditionId]));
$manager = $this->workflowManager->first();
```

Or...
https://gist.github.com/rbruhn/82811862cd3418b90282ee9b9a8381df#file-orderby
I needed for retrieving an earliest date from a table. It uses the above WhereFields too.
```
$this->transcription->pushCriteria(new WhereFields(['project_id' => 6, 'expedition_id' => 4]));
$this->transcription->pushCriteria(new OrderBy('finished_at'));
$record = $this->transcription->first();
```
I find making generic criteria removes the hassle of figuring out scope queries.

**Comment by rbruhn** [comment]
On second thought... as I go through my code looking at how this package will change things, I think there will be too much coupling. Especially when using criteria. For example, even the queries I used above would necessarily couple my code to this package. Imagine trying to swap out one repo for this one. I'd have to go back and change all these queries. I'm going to stick with my original repo where I had named methods so any new repo simply needs to duplicate those methods and alter it's own queries in the model itself.
